packages/components/toggle-button/src/toggle-button.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { expect, fixture } from '@open-wc/testing';
+import { a11ySnapshot, sendKeys } from '@web/test-runner-commands';
+import { html } from 'lit';
+import { spy } from 'sinon';
+import '../register.js';
+import { type ToggleButton } from './toggle-button.js';
+
+describe('sl-toggle-button', () => {
+  let el: ToggleButton;
+
+  describe('defaults', () => {
+    beforeEach(async () => {
+      el = await fixture(
+        html`<sl-toggle-button>
+          <sl-icon name="far-gear"></sl-icon>
+          <sl-icon name="fas-gear" slot="pressed"></sl-icon>
+        </sl-toggle-button>`
+      );
+    });
+
+    it('should render correctly', () => {
+      expect(el).shadowDom.to.equalSnapshot();
+    });
+
+    it('should have a button role', async () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-confusing-void-expression
+      const { role } = (await a11ySnapshot({ selector: 'sl-toggle-button' })) as any;
```

Comment:
`a11ySnapshot` is really useful, but unfortunately it's deprecated, with no replacement afaik